File: map-service/src/main/java/com/example/map/repository/RestaurantRepository.java
package com.example.map.repository;

import com.example.map.entity.Restaurant;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface RestaurantRepository extends JpaRepository<Restaurant, Long> {

    Optional<Restaurant> findByExternalId(String externalId);
}

File: apps/map-service/src/restaurant/restaurant.repository.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { Restaurant } from './restaurant.entity';

@Injectable()
export class RestaurantRepository {
  constructor(
    @InjectRepository(Restaurant)
    private readonly repo: Repository<Restaurant>,
  ) {}

  findByExternalId(externalId: string): Promise<Restaurant | null> {
    return this.repo.findOne({ where: { externalId } });
  }

  save(restaurant: Restaurant): Promise<Restaurant> {
    return this.repo.save(restaurant);
  }

}
